pkg/engine/internal/planner/logical/planner_test.go

```diff
@@ -292,6 +292,38 @@ func TestCanExecuteQuery(t *testing.T) {
 	}
 }
 
+func TestPlannerCreatesUnwrap(t *testing.T) {
+	t.Run("creates projection with unary cast operation instruction for metric query with unwrap duration", func(t *testing.T) {
+		// Query with duration unwrap in a sum_over_time metric query
+		q := &query{
+			statement: `sum by (status) (sum_over_time({app="api"} | unwrap duration(response_time) [5m]))`,
+			start:     3600,
+			end:       7200,
+			interval:  5 * time.Minute,
+		}
+
+		plan, err := BuildPlan(q)
+		require.NoError(t, err)
+		t.Logf("\n%s\n", plan.String())
+
+		// Assert against the correct SSA representation
+		// The UNWRAP should appear after SELECT operations but before RANGE_AGGREGATION
+		expected := `%1 = EQ label.app "api"
+%2 = MAKETABLE [selector=%1, predicates=[], shard=0_of_1]
+%3 = GTE builtin.timestamp 1970-01-01T00:55:00Z
+%4 = SELECT %2 [predicate=%3]
+%5 = LT builtin.timestamp 1970-01-01T02:00:00Z
+%6 = SELECT %4 [predicate=%5]
+%7 = PROJECT %6 [mode=E, expr=CAST_DURATION(ambiguous.response_time)]
```

Comment:
I think it's a small detail in how to denote the EXPAND=true, but personally I would also explicitly add ALL=true. (I did implement it that way in the DROP projection).

Not a strong opinion, though.

```suggestion
%7 = PROJECT %6 [mode=*E, expr=CAST_DURATION(ambiguous.response_time)]
```